osbuild/pipeline.py

```diff
@@ -314,6 +314,21 @@ def run(self, store, interactive=False, libdir=None, secrets=None):
         return results
 
 
+def detect_os():
```

Comment:
I'd love to see unit-tests that would contain os-release files from RHEL and Fedora and the expected output.

Reply:
Of course. Added a test for all current runners except fedora27.